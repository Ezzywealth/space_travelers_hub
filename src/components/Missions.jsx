import React from 'react';
import { useDispatch } from 'react-redux';
import asyncMission from '../redux/api/apiSlice';

const Missions = () => {
  const dispatch = useDispatch();
  return (
    <button type="button" onClick={() => dispatch(asyncMission())}>
      click
    </button>
  );
};
export default Missions;
