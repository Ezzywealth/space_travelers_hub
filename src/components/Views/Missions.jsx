import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import asyncMission from '../../redux/slices/api/apiSlice';
import Mission from './Mission';

const Missions = () => {
  const dispatch = useDispatch();
  const { missions } = useSelector((state) => state.missions);
  useEffect(() => {
    dispatch(asyncMission());
  }, [dispatch]);

  return (
    <>
      <table className="table-auto">
        <thead>
          <tr>
            <th className="border px-4 py-2">Mission</th>
            <th className="border px-4 py-2">Description</th>
            <th className="border px-4 py-2">Status</th>
            <th className="border px-4 py-2"> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <Mission
              key={mission.mission_id}
              missionId={mission.mission_id}
              missionName={mission.mission_name}
              description={mission.description}
              reserved={mission.reserved}
            />
          ))}
        </tbody>
      </table>
    </>
  );
};
export default Missions;
