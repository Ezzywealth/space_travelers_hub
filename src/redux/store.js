import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './slices/rockets/rocketSlice';
import missionReducer from './slices/missionSlice/missionSlice';
import menuReducer from './slices/menu/menuSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
    menu: menuReducer,
  },
});

export default store;
