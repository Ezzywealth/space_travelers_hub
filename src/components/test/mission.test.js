import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import TestRenderer, { act } from 'react-test-renderer';
import Missions from '../Views/Missions';
import store from '../../redux/store';

describe('Missions', () => {
  it('Snapshot of Missions Component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Table of Missions exist', () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const tableElement = screen.getByText(/Mission/i);
    expect(tableElement).toBeInTheDocument();
  });

  it('Join buttion', async () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const joinBtn = await screen.findAllByText('Join Mission');
    expect(joinBtn).toHaveLength(10);
  });

  it('Status exist', async () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const leaveBtn = await screen.findAllByText('NOT A MEMBER');
    expect(leaveBtn).toHaveLength(10);
  });

  it('Click on Join button', async () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    act(() => {
      const joinBtns = screen.getAllByText('Join Mission');
      joinBtns[0].click();
    });

    const status = await screen.getAllByText('Active Member');
    expect(status).toHaveLength(1);
  });

  it('Click on Leave button', async () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    act(() => {
      const leaveBtn = screen.getAllByText('Leave Mission');
      leaveBtn[0].click();
    });
    const status = await screen.getAllByText('NOT A MEMBER');
    expect(status).toHaveLength(10);
  });

  it('Click more than one Join button', async () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    act(() => {
      const joinBtns = screen.getAllByText('Join Mission');
      joinBtns[0].click();
    });

    act(() => {
      const joinBtns = screen.getAllByText('Join Mission');
      joinBtns[0].click();
    });

    const status = await screen.getAllByText('Active Member');
    expect(status).toHaveLength(2);
  });

  it('Click on more than one Leave button', async () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    act(() => {
      const leaveBtn = screen.getAllByText('Leave Mission');
      leaveBtn[0].click();
    });

    act(() => {
      const leaveBtn = screen.getAllByText('Leave Mission');
      leaveBtn[0].click();
    });

    const status = await screen.getAllByText('NOT A MEMBER');
    expect(status).toHaveLength(10);
  });
});
