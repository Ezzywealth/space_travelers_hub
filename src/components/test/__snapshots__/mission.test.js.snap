// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Render Missions Component 1`] = `
<main
  className="overflow-scroll"
>
  <table
    className="static table-auto mx-6"
  >
    <thead>
      <tr>
        <th
          className="border px-4 py-2"
        >
          Mission
        </th>
        <th
          className="border px-4 py-2"
        >
          Description
        </th>
        <th
          className="border px-4 py-2"
        >
          Status
        </th>
        <th
          className="border px-4 py-2"
        >
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</main>
`;

exports[`Missions Snapshot of Missions Component 1`] = `
<main
  className="overflow-scroll"
>
  <table
    className="static table-auto mx-6"
  >
    <thead>
      <tr>
        <th
          className="border px-4 py-2"
        >
          Mission
        </th>
        <th
          className="border px-4 py-2"
        >
          Description
        </th>
        <th
          className="border px-4 py-2"
        >
          Status
        </th>
        <th
          className="border px-4 py-2"
        >
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</main>
`;
